Hex code update from adding cancer_group as display_group (#1142)
AlexsLemonade/OpenPBTA-analysis: figures/mapping-histology-labels.Rmd

                display_order = as.numeric(display_group)) # save the factor order for text table export
```

# Make `cancer_group_order`

> `cancer_group` which is a shorter form of harmonised_diagnsosis with the following edits:

Reviewer: ```suggestion
`cancer_group` is a shorter form of `harmonized_diagnosis` with the following edits:
```

> - Removed Other, Benign tumor and Dysplasia/Gliosis from cancer_group

Reviewer: ```suggestion
- Removed Other, Benign tumor and Dysplasia/Gliosis from `cancer_group`
```

- Neurofibroma/Plexiform;Other updated to Neurofibroma/Plexiform
- Non-germinomatous germ cell tumor;Teratoma updated to Teratoma
- Anaplastic (malignant) meningioma, Meningothelial meningioma and Clear cell meningioma updated to Meningioma
- Embryonal Tumor with Multilayered Rosettes updated to Embryonal tumor with multilayer rosettes

Get ranks in order of big to small and make them into a new dataframe `cancer_group_order_df`. 

>  - We will remove any cancer_group with less than 5 sample counts

Reviewer: ```suggestion
```

Author: I did remove the filters so I'm resolving these, please reopen if I missed something

 - Remove any sample which has cancer_group==NA since those are mostly Normals or non CNS tumor samples belonging 
   to Other, Benign tumor and Dysplasia/Gliosis

>  - We will also replace "/" to ";" in cancer_group terms to work as filenames for outputs

Reviewer: ```suggestion
 - We will also replace "/" to " or " in cancer_group terms to work as filenames for outputs
```

This will look better in the legend

   
```{r}
cancer_group_order_df <- histology_table %>% 
  dplyr::count(cancer_group,name = "cancer_group_n") %>% 

>   # Keep only cancer_groups with 5 or more counts
>   dplyr::filter(cancer_group_n >=5,
>                 # remove NA
> 		!is.na(cancer_group)) %>% 

Reviewer: ```suggestion
  # Remove NA
  dplyr::filter(!is.na(cancer_group)) %>% 
```

Author: We are not removing the cancer_group NA now but keeping them and adding a gray color

  dplyr::mutate(
    # replace "/" with ";"

>     cancer_group = gsub("/",";",cancer_group),

Reviewer: ```suggestion
    cancer_group = gsub("/"," or ",cancer_group),
```
